src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -766,38 +766,47 @@ ss::future<> ntp_archiver::upload_until_term_change_legacy() {
-            // although we do also see this in practice on AWS S3 occasionally
-            // during normal operation.
-            vlog(
-              _rtclog.warn,
-              "Failed to upload {} segments out of {}",
-              non_compacted_upload_result.num_failed,
-              non_compacted_upload_result.num_succeeded
-                + non_compacted_upload_result.num_failed
-                + non_compacted_upload_result.num_cancelled);
-        } else if (non_compacted_upload_result.num_succeeded != 0) {
-            vlog(
-              _rtclog.debug,
-              "Successfully uploaded {} segments",
-              non_compacted_upload_result.num_succeeded);
+        bool uploads_paused
+          = !config::shard_local_cfg().cloud_storage_enable_segment_uploads();
+        std::optional<batch_result> result;
+        auto track_paused = _probe->register_archiver_on_hold(uploads_paused);
+        if (!uploads_paused) {
+            result = co_await upload_next_candidates(archival_stm_fence{
+              .read_write_fence = fence,
+              // Only use the rw-fence if the feature is enabled which requires
+              // major version upgrade.
+              .emit_rw_fence_cmd = emit_read_write_fence(_feature_table),
+            });
         }
+        if (result.has_value()) {
+            auto [compacted_upload_result, non_compacted_upload_result]
+              = result.value();
+            if (non_compacted_upload_result.num_failed != 0) {
+                // The logic in class `remote` already does retries: if we get
+                // here, it means the upload failed after several retries,
+                // justifying a warning to the operator: something non-transient
+                // may be wrong, although we do also see this in practice on AWS
+                // S3 occasionally during normal operation.
+                vlog(
+                  _rtclog.warn,
+                  "Failed to upload {} segments out of {}",
+                  non_compacted_upload_result.num_failed,
+                  non_compacted_upload_result.num_succeeded
+                    + non_compacted_upload_result.num_failed
+                    + non_compacted_upload_result.num_cancelled);
+            } else if (non_compacted_upload_result.num_succeeded != 0) {
+                vlog(
+                  _rtclog.debug,
+                  "Successfully uploaded {} segments",
+                  non_compacted_upload_result.num_succeeded);
+            }
 
-        if (non_compacted_upload_result.num_cancelled != 0) {
-            vlog(
-              _rtclog.debug,
-              "Cancelled upload of {} segments",
-              non_compacted_upload_result.num_cancelled);
+            if (non_compacted_upload_result.num_cancelled != 0) {
+                vlog(
+                  _rtclog.debug,
+                  "Cancelled upload of {} segments",
+                  non_compacted_upload_result.num_cancelled);
+            }
```

Comment:
when uploads_paused is true, then we fall through to what looks like the uploading of manifests. should those uploads be paused too?